benchmark/atoi-corpus.h

```diff
@@ -0,0 +1,135 @@
+#include "atoi.h"
+
+#include <vector>
+#include <string>
+#include <cstring>
+#include <random>
+
+struct Corpus8DecimalDigits {
+    std::vector<int> asNumbers_;
+    std::string characters_;
+
+    Corpus8DecimalDigits(std::vector<int> aNs, std::string cs):
+        asNumbers_(aNs),
+        characters_(cs)
+    {}
+
+    template<typename G>
+    static auto makeCorpus(G &generator) {
+        auto count = 1031; // 1031 is a prime number, this helps to disable in
+        // practice the branch predictor, the idea is to measure the performance
+        // of the code under measurement, not how the the unrealistic conditions
+        // of microbenchmarking help/hurt the code under measurement
+        std::string allCharacters;
+        allCharacters.resize(count * 9);
```

Comment:
why 9 and not a prime?

Reply:
The 9 arises from 8 bytes parsed and 1 null.  This is an artifact of my choice to use a function in the family of `printf` to convert from numbers to string, but I thought about it, and the 9 characters make things mostly unaligned, which I think is more representative of real uses.